cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -335,13 +335,14 @@ public void onSuccess(Object result) {
                     JSONObject jsonResult = (JSONObject) result;
                     consentUUID = jsonResult.getString("uuid");
                     metaData = jsonResult.getString("meta");
+                    userConsent = new GDPRUserConsent(jsonResult.getJSONObject("userConsent"));
+                    storeData();
                     if(jsonResult.has("msgJSON") && !jsonResult.isNull("msgJSON")) {
                         setNativeMessageView(jsonResult.getJSONObject("msgJSON"));
                         showView(nativeView);
                     } else if(jsonResult.has("url") && !jsonResult.isNull("url")){
                         loadConsentUI(jsonResult.getString("url"));
                     } else {
-                        userConsent = new GDPRUserConsent(jsonResult.getJSONObject("userConsent"));
```

Comment:
@leodoin `userConsent` was only being assigned if there was not `msgJSON` or `url`. So, only in the second launch that'd be the case.

Reply:
got it